src/bootstrap/src/ferrocene/dist.rs

```diff
@@ -323,3 +325,107 @@ impl Step for TestOutcomes {
         Some(tarball.generate())
     }
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub(crate) struct GenerateBuildMetadata;
+
+impl Step for GenerateBuildMetadata {
+    type Output = ();
+    const DEFAULT: bool = false;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("ferrocene-build-metadata")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(GenerateBuildMetadata);
+    }
+
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        let dist_dir = "build/dist";
```

Comment:
To avoid problems we should add at the top of the file:

```rust
if builder.config.dry_run() {
   return;
}
```

Reply:
The main concern I have around dry run is not the `--dry-run` flag (I haven't seen it much used in the wild), but that bootstrap will run steps twice, first with dry run and then without it. This means you're executing the `git` commands twice.

Reply:
Aight, changed it